### PR TITLE
Armor!

### PR DESCRIPTION
**Pull request by jsoconnell**
- New Armor class
- Monster attacks are weakened by armor defense
- Can now loot chests in inventory
- Minor cleanup

### CODE REVIEW

File: invadedhouse/run.py [hunk 1]
```diff
@@ -21,9 +20,9 @@
 def main():
     cprint('   ____                       __         __  __ __                   ', 'red')
     cprint('  /  _/___   _  __ ___  _ ___/ /___  ___/ / / // /___   __ __ ___  ___', 'yellow')
-    cprint(' _/ / / _ \ | |/ // _  `// _  // -_)/ _  / / _  // _ \ / // /(_-</  -_)', 'green')
-    cprint('/___//_//_/ |___/ \_,_/ \__,_/ __/ \_,_/  /_//_/ ___/ \__,_//___/ \__/ ', 'cyan')
-    print('')
+    cprint(' _/ / / _ \\ | |/ // _  `// _  // -_)/ _  / / _  // _ \\ / // /(_-</  -_)', 'green')
+    cprint('/___//_//_/ |___/ \\_,_/ \\__,_/ __/ \\_,_/  /_//_/ ___/ \\__,_//___/ \\__/ ', 'cyan')
+    print()
```
Comment (jsoconnell) [inline invadedhouse/run.py:25]: I know these lines look weird in the source code but it fixes bugs in parsing strings

Comment (redjoey) [reply invadedhouse/run.py:25]: why is the logo cut in half

File: invadedhouse/commands.py [hunk 2]
```diff
@@ -164,7 +170,7 @@ def interpret_command(self, cmd: str):
                 print('')
                 room_with_a_wall_that_can_be_smashed.set_exits([room_that_lets_you_into_the_sword_room, place_that_the_hole_leads_to])
         elif cmd in('give up', 'start over'):
-            cprint('Do you really want to give up? You\'re doing so good!, 'red')
+            cprint('Do you really want to give up? You\'re doing so good!', 'red')
```
Comment (redjoey) [inline invadedhouse/commands.py:173]: nice catch

File: invadedhouse/models/armor.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    ):
+        super().__init__(name, description)
+        self.dp = dp
+
+    def get_dp(self):
+        return self.dp
+
+    def look(self):
+        super().look()
+        print(f'Provides {self.get_dp()} points of protection.')
```
Comment (redjoey) [inline invadedhouse/models/armor.py:19]: armor!!!!!!!!!!! Yay

File: invadedhouse/models/player.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import sys
 from termcolor import cprint
 
+from .armor import Armor
```
Comment (redjoey) [inline invadedhouse/models/player.py:4]: Armor! YAAAY

File: invadedhouse/models/player.py [hunk 5]
```diff
@@ -72,15 +73,16 @@ def is_alive(self):
...
+            return
+
+        damage = min(monster.get_damage() - self.get_total_dp(), 0)
+        if damage > 0:
+            self.take_damage(damage)
+            print(f'You took {damage} damage!')
         else:
-            print(f'You have {self.get_hp()} of {self.get_max_hp()} HP remaining.')
-    
+            print(f'{monster.get_name()} tries to attack you, but they are no match for your armor.')
```
Comment (redjoey) [inline invadedhouse/models/player.py:84]: haha

File: invadedhouse/run.py [hunk 6]
```diff
@@ -3,7 +3,6 @@
 from commands import CommandInterpreter
 from models.house import House
 from models.player import Player
-from models.chest import Chest
```
Comment (redjoey) [inline invadedhouse/run.py:6]: okay? Nice?

File: invadedhouse/run.py (hunk 1 above)
Comment (redjoey) [inline invadedhouse/run.py:25]: HMMMMMMMMMMMMMMMMMMMMMMMMMM? WHAT? WHAT? WHAT THIS MEAN IT LOOK WIERD AND WHY DID YOU ADD IN THE DOUBLE SLAHSES AGAIN? ehh okay!

**Review by redjoey — APPROVED** [review]
nice!!!!

**Comment by redjoey** [comment]
Okay merge now

File: invadedhouse/run.py (hunk 1 above)
Comment (jsoconnell) [reply invadedhouse/run.py:25]: It's not... go [this view](https://github.com/redjoey/games/pull/12/files) to see the changes better

File: invadedhouse/models/player.py (hunk 5 above)
Comment (jsoconnell) [reply invadedhouse/models/player.py:84]: I knew you'd like that

File: invadedhouse/run.py (hunk 1 above)
Comment (jsoconnell) [reply invadedhouse/run.py:25]: the letter `\` is called an escape character... (remember how we use it for `\'` when we want to use quote characters? So to use it properly in a string you have to do what's called "double escape"... it'll look fine on screen when the program runs
